### DISCUSSION TITLE
BelongsTo FormData type changes when editing the resource

### DISCUSSION THREAD

**Discussion by SteffenAuer** [laravel/nova-issues#4383] [Q&A]
- Laravel Version: 8.83.15
- Nova Version: 4.7.0
- PHP Version: 8.0.19
- Database Driver & Version: MySQL 8.0.29
- Operating System and Version: Linux 5.10.104-linuxkit aarch64
- Browser type and version: Firefox 101 & Safari 15.5
- Reproduction Repository: (Private)

### Description:

- When used in dependent fields, the type of the value of a `BelongsTo` field differs between initial page load and when it is changed
   - On inital load, it is an `integer`
   - When the dependable field value is changed, it becomes a `string`

### Detailed steps to reproduce the issue on a fresh Nova installation:

```php
BelongsTo::make('Type')
    ->rules('required')
    ->withoutTrashed()
    ->filterable(),

Text::make('Offering')
    ->dependsOn('type', function (Text $field, NovaRequest $request, FormData $formData) {
        logger(gettype($formData->type)); // <<< HERE

        $type = $formData->type;

        // Note: $requiredType is a string value representing the id of a specific Type
        $type === $requiredType ? // <<< This check fails because the type differs
            $field->required() :
            $field->hide();
    })
    ->hideFromIndex(),
```

1. Create a resource with these fields
2. Open the edit page for that same resource
3. Change the value of the `Type` BelongsTo
4. In the logs, you'll see `integer` logged first and `string` logged second
   > [2022-06-08 01:06:44] local.DEBUG: integer  
   > [2022-06-08 01:06:48] local.DEBUG: string

**Comment by crynobone** [comment]
This can be caused by Eloquent casting the value to integer while normal XHR coming from `<select>` sending `string`. You should cover for both scenarios.
